zerver/actions/scheduled_messages.py

```diff
@@ -104,14 +108,15 @@ def do_schedule_messages(
                 scheduled_message.has_attachment = True
                 scheduled_message.save(update_fields=["has_attachment"])
 
-        event = {
-            "type": "scheduled_messages",
-            "op": "add",
-            "scheduled_messages": [
-                scheduled_message.to_dict() for scheduled_message, ignored in scheduled_messages
-            ],
-        }
-        send_event_on_commit(sender.realm, event, [sender.id])
+        if not skip_events:
+            event = {
+                "type": "scheduled_messages",
+                "op": "add",
+                "scheduled_messages": [
+                    scheduled_message.to_dict() for scheduled_message, ignored in scheduled_messages
+                ],
+            }
+            send_event_on_commit(sender.realm, event, [sender.id])
```

Comment:
What's the reason for skipping events?

Reply:
Sender is welcome bot here. No need to send the event to welcome bot.